sentinelhub/download/sentinelhub_client.py

```diff
@@ -50,28 +52,29 @@ def download(self, *args, **kwargs):
         finally:
             self.lock = None
 
-    @retry_temporal_errors
+    @retry_temporary_errors
     @fail_user_errors
     def _execute_download(self, request):
         """ Executes the download with a single thread and uses a rate limit object, which is shared between all threads
         """
-        thread_name = currentThread().getName()
-
         while True:
             sleep_time = self._execute_thread_safe(self.rate_limit.register_next)
 
             if sleep_time == 0:
+                LOGGER.debug('Sending %s request to %s with values %s',
+                             request.request_type.value, request.url, request.post_values)
                 response = self._do_download(request)
 
                 self._execute_thread_safe(self.rate_limit.update, response.headers)
 
                 if response.status_code != requests.status_codes.codes.TOO_MANY_REQUESTS:
                     response.raise_for_status()
 
-                    LOGGER.debug('%s: Successful download from %s', thread_name, request.url)
+                    LOGGER.debug('Successful %s request to %s', request.request_type.value, request.url)
                     return response.content
             else:
-                LOGGER.debug('%s: Sleeping for %0.2f', thread_name, sleep_time)
+                LOGGER.warning('Rate limit hit. Sleeping for %0.2f', sleep_time)
+                warnings.warn(f'Rate limit hit. Sleeping for {sleep_time:0.2f}', category=SHRuntimeWarning)
```

Comment:
I realized that this part is actually when `SentinelHubRateLimit` object says a request must wait. It is still ok that we log this, but I wouldn't raise a warning and I would change the log message. A warning about hitting a rate limit should be actually raised above in case where `response.status_code == requests.status_codes.codes.TOO_MANY_REQUESTS`.

Reply:
corrected and added a new warning class for rate limit